cmd/clusterctl/cmd/config_cluster.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type configClusterOptions struct {
+	kubeconfig             string
+	flavor                 string
+	bootstrapProvider      string
+	infrastructureProvider string
+
+	targetNamespace   string
+	kubernetesVersion string
+	controlplaneCount int
+	workerCount       int
+}
+
+var cc = &configClusterOptions{}
+
+var configClusterClusterCmd = &cobra.Command{
+	Use:   "cluster",
+	Short: "Generate templates for creating a Cluster API workload clusters",
+	Long: LongDesc(`
+		clusterctl ships with a list of well know providers; if necessary, edit
+		the $HOME/.clusterctl.yaml file to add new provider configurations or to customize existing ones.
```

Comment:
Is it safe to touch $HOME directly? Should we use a subdirectory in $HOME instead?

Reply:
Touching $HOME is not uncommon, but I don't have strong opinions

Reply:
Just based on personal experience it's hard for me to easily find things if it's just in $HOME. How about `$HOME/.cluster-api/clusterctl.yaml`?

Reply:
FWIW, both gcloud and github use `$HOME/.config/`, we could do `$HOME/.config/cluster-api/...`

Reply:
great! changed to $HOME/.config/cluster-api/...